src/controllers/mainController.ts

```diff
@@ -185,26 +186,40 @@ export default class MainController implements vscode.Disposable {
                 this._scriptingService = new ScriptingService(this._connectionMgr, this._vscodeWrapper);
                 this._context.subscriptions.push(
                     vscode.commands.registerCommand(
-                        Constants.cmdScriptSelect, async (node: TreeNodeInfo) => {
-                    let uri = await this._untitledSqlDocumentService.newQuery();
-                    let editor = this._vscodeWrapper.activeTextEditor;
-                    if (editor) {
+                    Constants.cmdScriptSelect, async (node: TreeNodeInfo) => {
+                    let actionPromise = new Promise<boolean>(async (resolve, reject) => {
+                        const nodeUri = ObjectExplorerUtils.getNodeUri(node);
```

Comment:
This doesn't seem to be need to be wrapped in a Promise - the method is already async so you could just have the body of the promise called here directly